stream-chat-android/src/main/java/com/getstream/sdk/chat/adapter/ChannelListItemAdapter.java

```diff
@@ -87,29 +87,27 @@ public int getItemViewType(int position) {
         // - if it extends baseChannelListItemView holder apply the click listeners and style;
         // - otherwise do nothing special
 
-        RecyclerView.ViewHolder anyViewHolder = viewHolderFactory.createChannelViewHolder(this, parent, viewType);
-
-        return anyViewHolder;
+        return viewHolderFactory.createChannelViewHolder(this, parent, viewType);
     }
 
     @Override
     public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
         Channel channelState = channels.get(position);
-        ((BaseChannelListItemViewHolder) holder).bind(this.context, channelState, position, null);
+        ((BaseChannelListItemViewHolder) holder).bind(channelState, position, FULL_CHANNEL_ITEM_PAYLOAD_DIFF);
     }
 
-    private ChannelItemPayloadDiff noDiff =  new ChannelItemPayloadDiff();
-
     @Override
     public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position, @NonNull List<Object> payloads) {
         Channel channelState = channels.get(position);
-        ChannelItemPayloadDiff diff;
+        ChannelItemPayloadDiff diff = EMPTY_CHANNEL_ITEM_PAYLOAD_DIFF;
         if (payloads.isEmpty()) {
-            diff = noDiff;
+            diff = FULL_CHANNEL_ITEM_PAYLOAD_DIFF;
         } else {
-            diff = (ChannelItemPayloadDiff) payloads.get(0);
+            for (int i = 0; i < payloads.size(); i++) {
+                diff = diff.plus((ChannelItemPayloadDiff) payloads.get(i));
+            }
```

Comment:
This would be so much nicer in Kotlin... Anyhow, well done, seems like a good change to get more precise with the diff.

Reply:
Yep, it is a `fold` call in Kotlin, but refactoring all class to Kotlin was out of the scope of this PR